channel_type.go

```diff
@@ -35,7 +35,7 @@ type (
 	modBehaviour string
 )
 
-type Permission struct {
+type ChannelTypePermission struct {
```

Comment:
Let's rename it to `Policy`  instead.

Reply:
@yaziine Policy is already used in this namespace
https://github.com/GetStream/stream-chat-go/blob/865e1919bb02070bf6373ea9bba0e30e37bd89ee/app.go#L77

Reply:
Actually we could get rid of this and use `Policy` directly because timestamps will be ignored by server to create and data shape is same

Reply:
Unfortunately `Action` seems to have different type in these classes 🤔

Reply:
let's address in a new PR, we need couple of helpers use these types